lib/internal/Magento/Framework/App/AreaList.php

```diff
@@ -72,7 +72,7 @@ public function getCodeByFrontName($frontName)
                 $resolver = $this->_resolverFactory->create($areaInfo['frontNameResolver']);
                 $areaInfo['frontName'] = $resolver->getFrontName(true);
             }
-            if ($areaInfo['frontName'] == $frontName) {
+            if ($areaInfo['frontName'] === $frontName) {
```

Comment:
As long as it's string - this actually does not matter.

Reply:
$frontname is not always string as described in the issue.